lib/spack/docs/packaging_guide.rst

```diff
@@ -2054,19 +2147,15 @@ The parameters are as follows:
     targets into.  It acts like a string, but it's actually its own
     special type, :py:class:`Prefix <spack.util.prefix.Prefix>`.
 
-``spec`` and ``prefix`` are passed to ``install`` for convenience.
-``spec`` is also available as an attribute on the package
-(``self.spec``), and ``prefix`` is actually an attribute of ``spec``
-(``spec.prefix``).
+The arguments ``spec`` and ``prefix`` are passed only for convenience, as they always
+correspond to ``self.spec`` and ``self.spec.prefix`` respectively.
```

Comment:
In `PackageBase`:
```python
@property
def prefix(self):
    """Get the prefix into which this package should be installed."""
    return self.spec.prefix
```
while phases get called like:
```python
getattr(self, phase)(self.spec, self.prefix)
```
So there should be no difference among the three.